### ISSUE TITLE
EVM.Verify gives an error in android device

### ISSUE THREAD

**Issue by yugansharora01** [ChainSafe/web3.unity#567]
**Describe the bug**
The bug is in the authentication process, EVM.Verify is giving an error. When the wallet authentication code runs and we get redirected to the metamask app ,after signing and clicking on the copy button when we move to the app again it gives an error (screenshot attached in the attachments). It works fine in unity simulator which means it works fine in desktop.

**To Reproduce**
1. Run the code below for authentication in android
2. See error in android logicat from EVM.Verify

```
async public void OnLogin()
    {
        // get current timestamp
        int timestamp = (int)(System.DateTime.UtcNow.Subtract(new System.DateTime(1970, 1, 1))).TotalSeconds;
        // set expiration time
        int expirationTime = timestamp + 60;
        // set message
        string message = expirationTime.ToString();
        // sign message
        string signature = await Web3Wallet.Sign(message);
        // verify account
        string account = await EVM.Verify(message, signature);
        int now = (int)(System.DateTime.UtcNow.Subtract(new System.DateTime(1970, 1, 1))).TotalSeconds;
        // validate
        if (account.Length == 42 && expirationTime >= now)
        {
            // save account
            PlayerPrefs.SetString("Account", account);
        }
}
```

**Screenshots**
![bug](https://github.com/ChainSafe/web3.unity/assets/75840050/dbc6adcf-9d23-48ae-833c-739c44d6dfd4)

**Desktop (please complete the following information):**
 - Chainsafe SDK version: [e.g. 1.0.6]
 - Unity Platform: [Windows, Mac, Linux or WebGL ]
 - Unity version [e.g. 2022] (Optional)
 - OS [e.g. Mac, Windows, Linux]

**Mobile (please complete the following information):**
 - Chainsafe SDK version: 1.6
 - Platform: Android
 - Type: Real device
 - OS version [e.g. 11.6] (Optional)

**Additional context**
It worked fine till last month probably , the integration of SDK was around January or February this year. I just opened my dashboard from https://dashboard.gaming.chainsafe.io/dashboard and just tried to edit the project and saw no platform was selected (maybe this option either not there when I created the project or it was optional) . I updated it but still it doesn't work.

**Comment by nftpixels** [comment]
Thanks for submitting this! 

I did some testing and I was able to replicate the issue. It's strange how it works just fine on desktop, but not Mobile. 

@yugansharora01 Have you tried another wallet, like TrustWallet to see if you get the same issue?

**Comment by yugansharora01** [comment]
@nftpixels Trust Wallet works on PC
but on android it just redirects to the metamask in play store

**Comment by sneakzttv** [comment]
this is fixed in 2.5.1
